src/modules/scraper/adapter/messaging/BlocksEventsConsumer.ts

```diff
@@ -204,52 +222,210 @@ export class BlocksEventsConsumer {
+    for (const txHash of Object.keys(events)) {
+      for (const matchedEvent of events[txHash]) {
+        try {
+          const { depositEvent, swapEvent, wethEvent } = matchedEvent;
+          const deposit = await this.fromFundsDepositedV3EventToDeposit(chainId, depositEvent, swapEvent, wethEvent);
+          const result = await this.depositRepository.insert(deposit);
+          await this.scraperQueuesService.publishMessage<BlockNumberQueueMessage>(ScraperQueue.BlockNumber, {
+            depositId: result.identifiers[0].id,
+          });
+        } catch (error) {
+          if (error instanceof QueryFailedError && error.driverError?.code === "23505") {
+            // Ignore duplicate key value violates unique constraint error.
+            this.logger.warn(error);
+          } else {
+            throw error;
+          }
+        }
+      }
     }
+  }
+
+  public matchDepositEventsWithSwapEvents(
+    depositEventsByTxHash: { [txHash: string]: FundsDepositedV3Event[] },
+    swapEventsByTxHash: { [txHash: string]: SwapBeforeBridgeEvent[] },
+    wethEventsByTxHash: { [txHash: string]: any[] },
+    chainId: number,
+  ) {
+    const events: { [txHash: string]: EnrichedDepositEvent[] } = {};
 
     for (const transactionHash of Object.keys(depositEventsByTxHash)) {
-      const pairs: [FundsDepositedV3Event, SwapBeforeBridgeEvent | undefined][] = [];
       const depositEvents = depositEventsByTxHash[transactionHash].sort((d1, d2) => d1.logIndex - d2.logIndex);
       const swapEvents = swapEventsByTxHash[transactionHash].sort((s1, s2) => s1.logIndex - s2.logIndex);
+      const wethEvents = wethEventsByTxHash[transactionHash].sort((s1, s2) => s1.logIndex - s2.logIndex);
 
       for (const depositEvent of depositEvents) {
+        const spokePoolAddresses = this.appConfig.values.web3.spokePoolContracts[chainId].map((sp) => sp.address);
+        const wethEventIndex = wethEvents.findIndex((e) => {
+          if (chainId === ChainIds.mainnet) {
+            const typedEvent = e as WethDepositEvent;
+            return (
+              e.logIndex < depositEvent.logIndex &&
+              typedEvent.args.wad.eq(depositEvent.args.inputAmount) &&
+              spokePoolAddresses.includes(typedEvent.args.dst)
+            );
+          } else if (chainId === ChainIds.optimism) {
+            const typedEvent = e as WethDepositEventOptimism;
+            return (
+              e.logIndex < depositEvent.logIndex &&
+              typedEvent.args.wad.eq(depositEvent.args.inputAmount) &&
+              spokePoolAddresses.includes(typedEvent.args.dst)
+            );
+          } else if (chainId === ChainIds.arbitrum) {
+            const typedEvent = e as WethTransfetEventArbitrum;
+            return (
+              e.logIndex < depositEvent.logIndex &&
+              typedEvent.args.value.eq(depositEvent.args.inputAmount) &&
+              spokePoolAddresses.includes(typedEvent.args.to)
+            );
```

Comment:
Don't we need to also add the check
```
typedEvent.args.from === ZERO_ADDRESS
```
? So that we don't consider a normal WETH transfer but rather a "deposit", i.e. mint.

Reply:
Yea, you are right. Should be very unlikely